tools/schemapi/utils.py

```diff
@@ -101,23 +101,47 @@ def add_literal(
                 tp = alias
         elif (alias := self._literals_invert.get(tp)) and replace:
             tp = alias
+        elif replace and info.is_union_enum():
+            # Handles one very specific edge case `WindowFieldDef`
```

Comment:
Special case referenced in comment

https://github.com/dangotbanned/altair/blob/ac23fd51460e69b1a7189da28b9a10e28eda2ac7/altair/vegalite/v5/schema/core.py#L26482